ArjunaJTS/jts/classes/com/arjuna/ats/internal/jts/orbspecific/interposition/coordinator/ServerTransaction.java

```diff
@@ -241,6 +241,9 @@ public final int doPhase2Commit ()
         }
 
 		super.phase2Commit(true);
+		if (super.failedList != null && super.failedList.size() > 0) {
+		    return ActionStatus.COMMITTING;
+		}
 
 		/*
 		 * Now do after completion stuff.
```

Comment:
A remark: if users have been relying on the old behaviour then their after completions won't run so would it be worthwhile highlighting the fix in the release notes? But I guess this is a corner case you are fixing in which case it probably doesn't need to be specifically brought out?